src/ray/core_worker/core_worker.h

```diff
@@ -1862,6 +1866,14 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   uint32_t pid_;
 
   absl::flat_hash_set<ObjectID> deleted_generator_ids_;
+
+  /// TODO(hjiang): Cached job runtime env info.
```

Comment:
@rynewang Caching job runtime env is not easy, since `OverrideRuntimeEnv` takes a job-based runtime env info and **updates** it in-place. I will leave a TODO for it for simplicity (this PR has been back-and-forth for so long), would like to merge it soon.